### PR TITLE
Tfrecords to parquet

### PR DESCRIPTION
**Pull request by bschifferer**
Example + function to convert tfrecords to parquet

### CODE REVIEW

**Comment by bschifferer** [comment]
@benfred updated to write each tfrecord to one parquet file by using cuDF
Issues with reading the parquet file with cuDF: https://github.com/rapidsai/cudf/issues/9216

File: nvtabular/framework_utils/tensorflow/__init__.py [hunk 1]
```diff
@@ -16,3 +16,4 @@
 
 # flake8: noqa
 from .feature_column_utils import make_feature_column_workflow
+from .tfrecords_to_parquet import convert_tfrecords_to_parquet
```
Comment (benfred) [inline nvtabular/framework_utils/tensorflow/__init__.py:19]: This adds a dependency on pandas_tfrecords - even if you just want to use the feature_column_utils. 

Maybe we can remove the re-import here - and just require people to import ```from nvtabular.framework_utils.tensorflow.tfrecords_to_parquet import convert_tfrecords_to_parquet``` if they want this functionality? 
```suggestion
```

File: nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                (), _get_feature_type(feature=feature), allow_missing=True
+            )
+
+    return features
+
+
+def _to_parquet(tfrecords, file, output_dir, chunks, convert_lists):
+    out = []
+    i = 0
+    j = 0
```
Comment (benfred) [inline nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py:78]: nitpick: I don't think 'j' is needed anymore here?
```suggestion
```

File: nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        out.append(row)
+        i += 1
+        if i == chunks:
+            df = cudf.DataFrame(out)
+            if convert_lists:
+                df = _convert_lists(df)
+            w.write_table(df)
+            i = 0
+            out = []
+            j += 1
```
Comment (benfred) [inline nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py:91]:
```suggestion
```

File: nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        Compression type of the tfrecords. Options: `""` (no compression), `"ZLIB"`, or `"GZIP"`
+    chunks: int
+        Chunks to convert tfrecords into parquet
+    convert_lists: Boolean
+        Output of tfrecords are lists. Set True to convert lists with fixed length to
+        individual columns in the output dataframe
+
+    """
+    # TODO: provide row_groupby_size parameter for parquet files
+    # TODO: optimize parquet files
```
Comment (benfred) [inline nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py:48]: I think by using the cudf ParquetWriter we should have optimized parquet files - with sensible rowgroup sizes (assuming chunks is large enough), maybe we can remove this TODO?  
```suggestion
```

File: examples/tensorflow/TFRecords-To-Parquet.ipynb [hunk 5]
```diff
@@ -0,0 +1,1139 @@
...
+       "4                              [0.945071, -0.071683]      0  "
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
```
Comment (benfred) [inline examples/tensorflow/TFRecords-To-Parquet.ipynb:1090]: We're already import pandas on line 182 - causing the linters to fail this PR with an error like 
```
Running flake8-nb

examples/tensorflow/TFRecords-To-Parquet.ipynb#In[22]:1:1: F811 redefinition of unused 'pd' from line 33
```


```suggestion
```

**Review by benfred — COMMENTED** [review]
This looks good! Left a couple minor comments below

File: nvtabular/framework_utils/tensorflow/__init__.py (hunk 1 above)
Comment (bschifferer) [reply nvtabular/framework_utils/tensorflow/__init__.py]: Ok, updated `__init__.py`

File: nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py (hunk 2 above)
Comment (bschifferer) [reply nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py]: Thanks, good point.

File: nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py (hunk 4 above)
Comment (bschifferer) [reply nvtabular/framework_utils/tensorflow/tfrecords_to_parquet.py]: Yes, I removed the ToDos

File: examples/tensorflow/TFRecords-To-Parquet.ipynb (hunk 5 above)
Comment (bschifferer) [reply examples/tensorflow/TFRecords-To-Parquet.ipynb]: I removed the 2nd import. It did not crash when I used flake8-nb locally.

**Comment by benfred** [comment]
rerun tests

**Review by benfred — APPROVED** [review]

**Pull request merged**